rmgpy/rmg/model.py

```diff
@@ -1443,22 +1443,19 @@ def add_reaction_to_core(self, rxn):
         """
         if rxn not in self.core.reactions:
             self.core.reactions.append(rxn)
-            if not self.core.phase_system.in_nose:
-                rms_species_list = self.core.phase_system.get_rms_species_list()
-                species_names = self.core.phase_system.get_species_names()
-                bits = np.array([spc.molecule[0].contains_surface_site() for spc in rxn.reactants+rxn.products])
-                in_edge = rxn in self.edge.reactions
-                if all(bits):
-                    self.core.phase_system.phases["Surface"].add_reaction(rxn)
-                    if not in_edge:
-                        self.edge.phase_system.phases["Surface"].add_reaction(rxn)
-                elif all(bits==False):
-                    self.core.phase_system.phases["Default"].add_reaction(rxn)
-                    if not in_edge:
-                        self.edge.phase_system.phases["Default"].add_reaction(rxn)
-                else:
-                    self.core.phase_system.interfaces[frozenset({"Default","Surface"})].add_reaction(rxn,species_names,rms_species_list)
-                    if not in_edge:
+            if rxn not in self.edge.reactions: 
+                #If a reaction is not in edge but is going to add to core, it is either a seed mechanism or a newly generated reaction where all reactants and products are already in core
+                #If the reaction is in edge, then the corresponding rms_rxn was moved from edge phase to core phase in pass_species already.
+                if not self.core.phase_system.in_nose:
+                    rms_species_list = self.core.phase_system.get_rms_species_list()
+                    species_names = self.core.phase_system.get_species_names()
+                    bits = np.array([spc.molecule[0].contains_surface_site() for spc in rxn.reactants+rxn.products])
+                    if all(bits):
+                        self.core.phase_system.phases["Surface"].add_reaction(rxn,self.edge.phase_system.phases["Surface"])
+                    elif all(bits==False):
+                        self.core.phase_system.phases["Default"].add_reaction(rxn,self.edge.phase_system.phases["Default"])
+                    else:
+                        self.core.phase_system.interfaces[frozenset({"Default","Surface"})].add_reaction(rxn,species_names,rms_species_list)
```

Comment:
I think I get whats going on here, but I found the commit message a bit difficult to understand.